ArduPlane/ArduPlane.cpp

```diff
@@ -963,6 +963,30 @@ bool Plane::flight_option_enabled(FlightOptions flight_option) const
     return g2.flight_options & flight_option;
 }
 
+#if AP_SCRIPTING_ENABLED
+// this implements the virtual function defined in AP_Vehicle for Plane to set the desired airspeed in m/s
+bool Plane::set_desired_airspeed(float airspeed_new)
+{
+    if (control_mode == &mode_guided || control_mode == &mode_auto) {
+        plane.new_airspeed_cm = constrain_float(airspeed_new, aparm.airspeed_min, aparm.airspeed_max) * 100.0f;
+        return true;
+    }
+    return false;
+}
+#endif
+
+#if AP_SCRIPTING_ENABLED
+// Helper function to let scripting set the guided mode loiter radius and direction
+bool Plane::set_guided_radius_and_direction(float radius, bool direction_is_ccw)
```

Comment:
I think we should remove or replace the "guided" part of the method name.  So perhaps change it to just, "set_radius_and_direction" or "set_circle_radius_and_direction" or "set_loiter_radius_and_direction".

Reply:
This is implementing a virtual method that is defined in AP_Vehicle.h. I'm thinking we wouldn't want to change that.

Reply:
The thing is it really only sets the radius and direction _for guided mode_ - it doesn't set the loiter radius/direction for many other places in plane, only the specific one used for guided mode. 
The way the guided mode radius_direction works is it is an override for the general loiter behavior. If not set then guided mode will use the general loiter values, but if set, they will apply for guided mode only.

Reply:
I've removed this in favor of Tridge's MAVLink Api suggestion #37265